jobs/build/rhcos_sync/S3-rhcossync.sh

```diff
@@ -0,0 +1,191 @@
+#!/bin/bash
+set -exo pipefail
+
+# Where to put this on the s3 mirror, such as '4.2' or 'pre-release':
+RHCOS_MIRROR_PREFIX=
+ARCH=x86_64
+# RHCOS version, like 42.80.20190828.2:
+BUILDID=
+# release version, like 4.2.0:
+VERSION=
+FORCE=0
+TEST=0
+BASEDIR=
+NOLATEST=0
+NOMIRROR=0
+
+function usage() {
+    cat <<EOF
+usage: ${0} [OPTIONS]
+
+This script will create directories on the s3 mirror under
+/pub/openshift-v4/x86_64/dependencies/rhcos/{--PREFIX}/{--VERSION}/
+containing the items provided in --synclist as well as a sha256sum.txt
+file generated from those items.
+
+Required Options:
+
+  --version        The RHCOS version to mirror (ex: 4.2.0, 4.2.0-0.nightly-2019-08-28-152644)
+  --prefix         The parent directory to mirror to (ex: 4.1, 4.2, pre-release)
+  --synclist       Path to the file of items (URLs) to mirror (whitespace separated)
+  --basedir        Base filesystem path in which the --PREFIX/--VERSION directories exist
+
+Optional Options:
+
+  --nolatest       Do not update the 'latest' symlink after downloading
+  --test           Test inputs, but ensure nothing can ever go out to the mirrors
+  --nomirror       Do not run the push.pub script after downloading
+
+
+EOF
+}
+
+function downloadImages() {
```

Comment:
I see the necessity for downloading this again - other download is on use-mirror. but my experience with downloading this on buildvm previously was that it's unbearably slow. have you tried?

if there's a way to temp use s3 creds on use-mirror that _might_ be preferable but i hope i'm wrong because that doesn't seem like a good architecture. (and as far as i know, copying directly from one bucket to another would not improve it any)

Reply:
I tested copying from s3 to s3 and it was significantly faster from my laptop. So we have that going for us. 
I don't expect use-mirror to be around much longer, so we would need to provide our own staging location.